internal/metrics/metrics_test.go

```diff
@@ -0,0 +1,393 @@
+				"OTEL_METRICS_EXPORTER": "prometheus",
+			},
+			expectedConsoleContains: "",
+			expectServiceName:       "",
+			expectResource:          false,
+		},
+		{
+			name: "no console output when disabled",
+			env: map[string]string{
+				"OTEL_METRICS_EXPORTER": "none",
+			},
+			expectedConsoleContains: "",
+			expectServiceName:       "",
+			expectResource:          false,
+		},
+		{
+			name: "no console output when SDK disabled",
+			env: map[string]string{
+				"OTEL_SDK_DISABLED":     "true",
+				"OTEL_METRICS_EXPORTER": "console",
+			},
+			expectedConsoleContains: "",
+			expectServiceName:       "",
+			expectResource:          false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			synctest.Test(t, func(t *testing.T) {
+				t.Helper()
+				t.Setenv("OTEL_METRIC_EXPORT_INTERVAL", "100")
+				for k, v := range tt.env {
+					t.Setenv(k, v)
+				}
+
+				var stdout bytes.Buffer
+				manualReader := sdkmetric.NewManualReader()
+
+				meter, shutdown, err := NewMetricsFromEnv(t.Context(), &stdout, manualReader)
+				require.NoError(t, err)
+				require.NotNil(t, meter)
+				require.NotNil(t, shutdown)
+				defer func() {
+					_ = shutdown(context.Background())
+				}()
+
+				// Create and record a metric
+				counter, err := meter.Int64Counter("test.console.metric",
+					metric.WithDescription("A test metric"),
+					metric.WithUnit("1"))
+				require.NoError(t, err)
+				counter.Add(t.Context(), 42)
+
+				// Collect metrics via Prometheus reader
+				var rm metricdata.ResourceMetrics
+				err = manualReader.Collect(t.Context(), &rm)
+				require.NoError(t, err)
+				require.NotEmpty(t, rm.ScopeMetrics, "Prometheus reader should collect metrics")
+
+				// Verify resource attributes
+				found := false
+				var serviceName string
+				for _, attr := range rm.Resource.Attributes() {
+					if attr.Key == "service.name" {
+						found = true
+						serviceName = attr.Value.AsString()
+						break
+					}
+				}
+				if tt.expectResource {
+					require.True(t, found, "service.name attribute should be present")
+					require.Equal(t, tt.expectServiceName, serviceName)
+				}
+
+				// Check console output
+				if tt.expectedConsoleContains != "" {
+					time.Sleep(150 * time.Millisecond)
+					synctest.Wait()
+					output := stdout.String()
+					// Single check for all expected content in console output
+					expectedParts := []string{tt.expectedConsoleContains, "42"}
+					if tt.expectServiceName != "" {
+						expectedParts = append(expectedParts, tt.expectServiceName)
+					}
+					for _, part := range expectedParts {
+						require.Contains(t, output, part)
+					}
+				} else {
+					output := stdout.String()
+					require.Empty(t, output, "no console output expected")
+				}
+			})
+		})
+	}
+}
+
+// TestNewMetricsFromEnv_NetworkExporters tests OTLP and other network-based exporters.
+// We CANNOT use synctest here because it creates a "bubble" where goroutines are isolated
```

Comment:
ps I intentionally left this in as it is something I will discuss in the golang meetup tomorrow ;)